bundles/org.openhab.binding.pilight/src/main/java/org/openhab/binding/pilight/internal/PilightConnector.java

```diff
@@ -225,14 +231,17 @@ private void connect() throws InterruptedException {
      */
     public void sendAction(Action action) {
         delayedActionQueue.add(action);
+        final @Nullable ScheduledFuture<?> delayedActionWorkerFuture = this.delayedActionWorkerFuture;
 
         if (delayedActionWorkerFuture == null || delayedActionWorkerFuture.isCancelled()) {
-            delayedActionWorkerFuture = scheduler.scheduleWithFixedDelay(() -> {
+            this.delayedActionWorkerFuture = scheduler.scheduleWithFixedDelay(() -> {
                 if (!delayedActionQueue.isEmpty()) {
                     doSendAction(delayedActionQueue.poll());
                 } else {
-                    delayedActionWorkerFuture.cancel(false);
-                    delayedActionWorkerFuture = null;
+                    if (delayedActionWorkerFuture != null) {
+                        delayedActionWorkerFuture.cancel(false);
+                    }
```

Comment:
This would cancel the already canceled future.

Reply:
I'm not quite sure, what you mean. This call takes place in the lambda function of the fixed delay scheduler. In each run of it, I check if there are remaining elements in the `delayedActionQueue` and poll one of them to call `doSendAction()`. If the queue is empty I need to cancel the scheduler. I think it is a valid case here or am I wrong?
Maybe you can explain in more detail what you are thinking?

Reply:
I'm saying that the old code referenced the `delayedActionWorkerFuture` field, but now it is referencing the `delayedActionWorkerFuture` local variable. The local variable has already been determined to be cancelled at this point in the code. It clear that your intent is to cancel the future field rather than the local variable. You need to make a second local variable inside the lambda to grab a fresh reference to the future.

Reply:
Ah sure, now I get your point!